source/slangc/main.cpp

```diff
@@ -55,47 +55,40 @@ static SlangResult innerMain(int argc, char** argv)
         }
     }
 
+    SlangResult res = SLANG_OK;
+
 #ifndef _DEBUG
     try
 #endif
     {
-        // Run the compiler (this will produce any diagnostics through
-        // our callback above).
-        if (SLANG_FAILED(spCompile(compileRequest)))
-        {
-            // If the compilation failed, then get out of here...
-            // Turn into an internal Result -> such that return code can be used to vary result to match previous behavior
-            return SLANG_E_INTERNAL_COMPILE_FAILED;
-        }
-
-        // Now that we are done, clean up after ourselves
-
-        spDestroyCompileRequest(compileRequest);
-        spDestroySession(session);
+        // Run the compiler (this will produce any diagnostics through SLANG_WRITER_TARGET_TYPE_DIAGNOSTIC).
+        res = spCompile(compileRequest);
+        // If the compilation failed, then get out of here...
+        // Turn into an internal Result -> such that return code can be used to vary result to match previous behavior
+        res = SLANG_FAILED(res) ? SLANG_E_INTERNAL_FAIL : res;
     }
 #ifndef _DEBUG
     catch (Exception & e)
     {
-        printf("internal compiler error: %S\n", e.Message.ToWString().begin());
-        return SLANG_FAIL;
+        AppContext::getStdOut().print("internal compiler error: %S\n", e.Message.ToWString().begin());
+        res = SLANG_FAIL;
     }
 #endif
-    return SLANG_OK;    
+
+    // Now that we are done, clean up after ourselves
+    spDestroyCompileRequest(compileRequest);
+    return res;
 }
 
 int MAIN(int argc, char** argv)
 {
```

Comment:
It is also a bit silly that on Windows we now have `wmain` that calls `MAIN` that calls `innerMain`. There's not a great way around it, and each of the layers of nesting is solving a different problem.